### ISSUE TITLE
Use Gradle `changing` to determine snapshots

### ISSUE THREAD

**Issue by briandealwis** [GoogleContainerTools/jib#694]
From 
https://github.com/GoogleContainerTools/jib/pull/584#discussion_r204366857.  Gradle supports [explicitly indicating that a dependency may be changing](
https://stackoverflow.com/a/42058780/600339) and [provides an API](https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/ExternalModuleDependency.html#isChanging--) to check.  The [docs](https://docs.gradle.org/current/userguide/declaring_dependencies.html#sub:declaring_dependency_with_changing_version) [hint](https://docs.gradle.org/current/userguide/troubleshooting_dependency_resolution.html#sub:dynamic_versions_and_changing_modules) that SNAPSHOTs are marked as changing.

**Comment by chanseokoh** [comment]
> The [docs](https://docs.gradle.org/current/userguide/declaring_dependencies.html#sub:declaring_dependency_with_changing_version) [hint](https://docs.gradle.org/current/userguide/troubleshooting_dependency_resolution.html#sub:dynamic_versions_and_changing_modules) that SNAPSHOTs are marked as changing.

It does seem like dependencies with the `-SNAPSHOT` suffix will get the `changing` set to true, but it'd be always good to explicitly verify it.

**Comment by TadCordle** [comment]
Clearing the milestone on this - turns out to not be as easy as we thought (using `isChanging()` means we need to find the `ResolvedDependency` object associated with each jar file, which doesn't seem to be very straightforward).

**Comment by mcabezas** [comment]
Hi every one, I would like to contribute on this, is this issue still open?

**Comment by mcabezas** [comment]
Hi everyone, I would like to contribute on this, is this issue still open?

**Comment by chanseokoh** [comment]
Yes, the issue is still open. We would greatly appreciate your help. Maybe both 1) dependencies marked changing; and 2) dependencies whose filename ends with `SNAPSHOT` should be added to the snapshot layer.

**Comment by loosebazooka** [comment]
Hey @mcabezas this actually requires some underlying changes to where dependent layering is processed. If you look at `JavaContainerBuilder` you'll see that splitting of snapshot and non-snapshot dependencies is done after gradle has handed off its list of dependencies. We need to update that api first before you will have the tools to handle this.

**Comment by mcabezas** [comment]
@chanseokoh Thanks to clarify the scope of the change. I'm justo looking to retrieve something to this awesome tol builder.

@loosebazooka So may I wait for the API  update before starting? I would like to contribute with something let me know what can I choose as first contribution. I'm open to your suggestions.

**Comment by loosebazooka** [comment]
Hrmm... The discussion is happening here: #1724 

I'm not sure if this is the ideal change to tackle as a contributor as I expect the plugins' dependency filtering to happen with the api change. However if I see an opportunity for contribution I will update the bug.

**Comment by chanseokoh** [comment]
The planned underlying API changes and dependency filtering is complete. This issue is again a "good first issue" for contribution.

**Comment by loosebazooka** [comment]
Ah right, @mcabezas, you can take a look again if you want.

Most of the relevant code is in https://github.com/GoogleContainerTools/jib/blob/master/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

**Comment by rajovictor** [comment]
@chanseokoh I created a PR #2078
